### PR TITLE
Adding Actionlog Privacy plugin

### PR DESCRIPTION
**Pull request by Hackwar**
This exports the actionlog data. This currently does not export actionlog entries where the user is the object of the action and not the subject. It also does not sanitise the output of other peoples data (simply because we don't know what someone might log). So if an admin enables another user, that log entry should be in the other users exported log and the admin should be anonymised. Unfortunately the system does not allow this (yet). 

We are slowly getting to much plugins for numberspace in our extension table...

### CODE REVIEW

**Comment by Hackwar** [comment]
This solves #160

**Comment by brianteeman** [comment]
when used in language strings etc actionlog should be TWO words

File: installation/sql/sqlazure/joomla.sql [hunk 1]
```diff
@@ -890,6 +890,7 @@ INSERT INTO "#__extensions" ("extension_id", "package_id", "name", "type", "elem
 (487, 0, 'plg_privacy_user', 'plugin', 'user', 'privacy', 0, 1, 1, 0, '', '{}', '', '', 0, '1900-01-01 00:00:00', 0, 0),
 (488, 0, 'plg_quickicon_privacycheck', 'plugin', 'privacycheck', 'quickicon', 0, 1, 1, 0, '', '{}', '', '', 0, '1900-01-01 00:00:00', 0, 0),
 (489, 0, 'plg_user_terms', 'plugin', 'terms', 'user', 0, 0, 1, 0, '', '{}', '', '', 0, '1900-01-01 00:00:00', 0, 0),
+(490, 0, 'plg_privacy_actionlogs', 'plugin', 'actionlogs', 'privacy', 0, 0, 1, 0, '', '{}', '', '', 0, '0000-00-00 00:00:00', 0, 0),
```
Comment (alikon) [inline installation/sql/sqlazure/joomla.sql]: wrong 0000-00-00 00:00:00 mssql

File: installation/sql/postgresql/joomla.sql [hunk 2]
```diff
@@ -675,6 +675,7 @@ INSERT INTO "#__extensions" ("extension_id", "package_id", "name", "type", "elem
 (487, 0, 'plg_privacy_user', 'plugin', 'user', 'privacy', 0, 1, 1, 0, '', '{}', '', '', 0, '1970-01-01 00:00:00', 0, 0),
 (488, 0, 'plg_quickicon_privacycheck', 'plugin', 'privacycheck', 'quickicon', 0, 1, 1, 0, '', '{}', '', '', 0, '1970-01-01 00:00:00', 0, 0),
 (489, 0, 'plg_user_terms', 'plugin', 'terms', 'user', 0, 0, 1, 0, '', '{}', '', '', 0, '1970-01-01 00:00:00', 0, 0),
+(490, 0, 'plg_privacy_actionlogs', 'plugin', 'actionlogs', 'privacy', 0, 0, 1, 0, '', '{}', '', '', 0, '0000-00-00 00:00:00', 0, 0),
```
Comment (alikon) [inline installation/sql/postgresql/joomla.sql]: wrong 0000-00-00 00:00:00 postgresql

File: administrator/components/com_admin/sql/updates/mysql/3.9.0-2018-07-11.sql [hunk 3]
```diff
@@ -0,0 +1,2 @@
+INSERT INTO `#__extensions` (`extension_id`, `package_id`, `name`, `type`, `element`, `folder`, `client_id`, `enabled`, `access`, `protected`, `manifest_cache`, `params`, `custom_data`, `system_data`, `checked_out`, `checked_out_time`, `ordering`, `state`) VALUES
+(490, 0, 'plg_privacy_actionlogs', 'plugin', 'actionlogs', 'privacy', 0, 0, 1, 0, '', '{}', '', '', 0, '0000-00-00 00:00:00', 0, 0);
```
Comment (mbabker) [inline administrator/components/com_admin/sql/updates/mysql/3.9.0-2018-07-11.sql]: ID 493 after merging the other plugins.

**Comment by Hackwar** [comment]
I did the changes for the SQLs. Should be ready to go now. :smile:

**Pull request merged**